backends-velox/src/main/scala/org/apache/spark/shuffle/VeloxColumnarShuffleWriter.scala

```diff
@@ -162,7 +161,8 @@ class VeloxColumnarShuffleWriter[K, V](
           .split(nativeSplitter, cb.numRows, cArray.memoryAddress())
         dep.splitTime.add(System.nanoTime() - startTime)
         dep.numInputRows.add(cb.numRows)
-        writeMetrics.incRecordsWritten(1)
+        // This metric is important, AQE use it to decide if EliminateLimit
+        writeMetrics.incRecordsWritten(cb.numRows())
         cArray.close()
```

Comment:
@jinchengchenghh 
Can you add one metric named "input_batches" to count the number of record batch number? The metric is used in our eventlog analysis.

Reply:
ok, I will start a new PR